### ISSUE TITLE
add deterministic tests and test vectors

### ISSUE THREAD

**Issue by david415** [cryptix/secretstream#1]
"other language implementations of secret-handshake should share these test vectors. this would give implementers some confidence that they've done it correctly. additionally the binary compatibility among language implementations might prove useful."

https://github.com/auditdrivencrypto/secret-handshake/issues/5

**Comment by cryptix** [comment]
agreed! would synchronize this with auditdrivencrypto/secret-handshake#6
